### DISCUSSION TITLE
Age issues (not me :) )

### DISCUSSION THREAD

**Discussion by raymond-w** [djong1/gp3s#102] [General]
https://www.gps-speedsurfing.com/default.aspx?mnu=rankings&Year=-&Tab=0&Gender=0&WeightClass=0&AgeClass=1&SpeedType=0

There are a lot of riders without a proper age.
They are now in the 0-10/11-14 ranking which now is not usable for the kids.

Maybe add a filter to check registration date and user age.

**Comment by djong1** [comment]
The problem is that users explicitly screw up by providing the wrong date of birth.  The best way I see for now is flagging the session, what do you think?

**Comment by jhdebruin** [comment]
The problem is that users explicitly screw up by providing the wrong date of birth. The best way I see for now is flagging the session, what do you think? @raymondwortel

**Reply by raymond-w** [reply]
had to think about it :)

**Comment by raymond-w** [comment]
I think it needs better filtering
Example : https://www.gps-speedsurfing.com/default.aspx?mnu=user&smnu=sessions&uid=23062
The age class clearly indicates Master (35-50)
But the session is shown in the Youth ranking : https://www.gps-speedsurfing.com/default.aspx?mnu=rankings&Year=-&Tab=0&Gender=0&WeightClass=0&AgeClass=1&SpeedType=0

Flagging ALL the sessions will be insane, how much session where posted before the age was set properly?
10 sessions per rider? 100? 10.000 on the whole site?
And do we need to flag every session? Or will the site have a check for the next session?

The solution would be a script that checks age and dates
If the difference of the first session and the age of the rider is <10y AND speed is above xx KTS, the chances are almost 100% it is not correct.
Also, if the rider later changed the age in a correct value we could use that value.
Save the outcome in an age-verified db field, if you rerun the script in the future it can skip already checked users.

/edit
That age-verified field could also be used when an user wants to post the first session.
Display a popup the need to check there profile fields and check settings like age etc.
After that the could get an age-verified checkmark

**Comment by raymond-w** [comment]
@jhdebruin @djong1 
Any updates around this?
